### PR TITLE
Rework examples to be privacy preserving

### PR DESCRIPTION
**Pull request by tlodderstedt**
addresses issue #16

### CODE REVIEW

File: main.md [hunk 1]
```diff
@@ -288,7 +283,30 @@ In the context of a request object as specified in [@I-D.ietf-oauth-jwsreq], `au
 
 Note: Authorization request URIs containing authorization details in a request parameter or a request object can become very long. Implementers SHOULD therefore consider using the `request_uri` parameter as defined in [@I-D.ietf-oauth-jwsreq], potentially in combination with the pushed request object mechanism as defined in [@I-D.lodderstedt-oauth-par] to pass authorization details in a reliable and secure manner.
 
-## Authorization Request Processing 
+Here is an example of a pushed authorization request that sends the authorization request data directly to the AS via a HTTPS-protected connection:
```
Comment (jricher) [inline main.md]: Add an informative reference to the PAR draft here.

File: main.md [hunk 2]
```diff
@@ -236,21 +235,17 @@ GET /authorize?response_type=code&client_id=s6BhdRkqt3
...
-    %22amount%22%3A%22123.50%22%7D%2C%22debtorAccount%22%3A%7B%
-    22iban%22%3A%22DE40100100103307118608%22%7D%2C%22creditorNa
-    me%22%3A%22Merchant123%22%2C%22creditorAccount%22%3A%7B%22i
-    ban%22%3A%22DE02100100109307118603%22%7D%2C%22remittanceInf
-    ormationUnstructured%22%3A%22Ref%20Number%20Merchant%22%7D%
-    5D%0A%20%20%20 HTTP/1.1
+    &authorization_details=%5B%7B%22type%22:%22account_information
+    %22,%22actions%22:%5B%22list_accounts%22,%22read_balances%22,%
+    22read_transactions%22%5D,%22locations%22:%5B%22https://exampl
+    e.com/accounts%22%5D%7D%5D HTTP/1.1
```
Comment (b---c) [inline main.md]: Looks like the full authz deets JSON is included now. So the `(JSON string trimmed for brevity)` part could and should be removed from the text just before this example.

File: main.md [hunk 3]
```diff
@@ -236,21 +235,17 @@ GET /authorize?response_type=code&client_id=s6BhdRkqt3
...
-    B%22initiate%22%2C%22status%22%2C%22cancel%22%5D%2C%22locat
-    ions%22%3A%5B%22https%3A%2F%2Fexample.com%2Fpayments%22%5D%
-    2C%22instructedAmount%22%3A%7B%22currency%22%3A%22EUR%22%2C
-    %22amount%22%3A%22123.50%22%7D%2C%22debtorAccount%22%3A%7B%
-    22iban%22%3A%22DE40100100103307118608%22%7D%2C%22creditorNa
-    me%22%3A%22Merchant123%22%2C%22creditorAccount%22%3A%7B%22i
-    ban%22%3A%22DE02100100109307118603%22%7D%2C%22remittanceInf
-    ormationUnstructured%22%3A%22Ref%20Number%20Merchant%22%7D%
-    5D%0A%20%20%20 HTTP/1.1
+    &authorization_details=%5B%7B%22type%22:%22account_information
```
Comment (b---c) [inline main.md]: I'll admit that the exact rules for what characters need to be encoded with application/x-www-form-urlencoded are a bit mysterious to me. But I think that `/` and `:` are among them and those are unencoded here (and elsewhere). 

https://www.urldecoder.org/ gives the below and Java's `URLEncoder.encode(...)` agrees with that.  
`%5B%7B%22type%22%3A%22account_information%22%2C%22actions%22%3A%5B%22list_accounts%22%2C%22read_balances%22%2C%22read_transactions%22%5D%2C%22locations%22%3A%5B%22https%3A%2F%2Fexample.com%2Faccounts%22%5D%7D%5D`

File: main.md (hunk 1 above)
Comment (b---c) [reply main.md]: There is a PAR ref in the previous paragraph. You think it's needed here too?

File: main.md [hunk 4]
```diff
@@ -288,7 +283,30 @@ In the context of a request object as specified in [@I-D.ietf-oauth-jwsreq], `au
...
+  Host: as.example.com
+  Content-Type: application/x-www-form-urlencoded
+  Authorization: Basic czZCaGRSa3F0Mzo3RmpmcDBaQnIxS3REUmJuZlZkbUl3
+
+  response_type=code&
+  client_id=s6BhdRkqt3
+  &state=af0ifjsldkj
+  &redirect_uri=https%3A%2F%2Fclient.example.org%2Fcb 
+  &code_challenge_method=S256
+  &code_challenge=%5B%7B%22type%22:%22https://www.someorg.compayment_ini
```
Comment (b---c) [inline main.md]: the code_challenge value got cut off or something along with the authorization_details parameter name. 

`&code_challenge=K2-ltc83acc4h0c9w6ESC_rEMTJ3bww-uCHaoeK1t8U
  &authorization_details=%5B%7B%22 ... `

File: main.md [hunk 5]
```diff
@@ -288,7 +283,30 @@ In the context of a request object as specified in [@I-D.ietf-oauth-jwsreq], `au
...
+  tiation%22,%22actions%22:%5B%22initiate%22,%22status%22,%22cancel%22%5
+  D,%22locations%22:%5B%22https://example.com/payments%22%5D,%22instruct
```
Comment (b---c) [inline main.md]: URL encoding of slash and colon here too?

File: main.md (hunk 1 above)
Comment (tlodderstedt) [reply main.md]: I merged both paragraphs to make the single PAR reference unambiguous.

File: main.md (hunk 2 above)
Comment (tlodderstedt) [reply main.md]: done

File: main.md (hunk 3 above)
Comment (tlodderstedt) [reply main.md]: I couldn't find a definitiv answer whether / and : must be encoded but it's an option. I changed the examples to use this encoding.

Comment (b---c) [reply main.md]: wfm

**Pull request merged**